src/ini.rs

```diff
@@ -851,3 +854,34 @@ impl Ini {
         self.map.get_mut(&section)?.remove(&key)
     }
 }
+
+#[cfg(feature = "async-std")]
+impl Ini {
+    ///Loads a file asynchronously from a defined path, parses it and puts the hashmap into our struct.
+    ///At one time, it only stores one configuration, so each call to `load()` or `read()` will clear the existing `Map`, if present.
+    ///
+    ///Usage is similar to `load`, but `.await` must be called after along with the usual async rules.
+    ///
+    ///Returns `Ok(map)` with a clone of the stored `Map` if no errors are thrown or else `Err(error_string)`.
+    ///Use `get_mut_map()` if you want a mutable reference.
+    pub async fn load_async<T: AsRef<AsyncPath>>(
+        &mut self,
+        path: T,
+    ) -> Result<Map<String, Map<String, Option<String>>>, String> {
+        let s = async_fs::read_to_string(&path)
```

Comment:
We don't need to assign the variable here, something like:
```rust
        self.map = match self.parse(match async_fs::read_to_string(&path).await {
            Err(why) => return Err(format!(
                    "couldn't read {}: {}",
                    &path.as_ref().display(),
                    why
                )),
            Ok(s) => s,
            }) {
                Err(why) => {
                    return Err(format!(
                        "couldn't read {}: {}",
                        &path.as_ref().display(),
                        why
                    ))
                }
                Ok(map) => map,
        };
```
I've benchmarked the synchronous version, it's a bit faster (~1 microsecond), async benching is a bit too much of a headache.

Reply:
Assigning to a variable is more readable, in my opinion. Especially when mapping error values. It's your project, though, I'll change it.